index.bs

```diff
@@ -175,6 +175,88 @@ partial interface Navigator {
 };
 </pre>
 
+# Audio session algorithms # {#audio-session-algorithms}
+
+## Update AudioSession's type ## {#audio-session-update-type-algorithm}
+
+To <dfn>update the type</dfn> of |audioSession|, the user agent MUST run the following steps:
+1. If |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] is `true`, abort these steps.
+1. Set |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] to `true`.
+1. [=Queue a task=] to run the following steps:
+    1. Set |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] to `false`.
+    1. If |audioSession|.[=AudioSession/[[type]]=] is the same as |audioSession|.[=AudioSession/[[appliedType]]=], abort these steps.
+    1. Set |audioSession|.[=AudioSession/[[appliedType]]=] to |audioSession|.[=AudioSession/[[type]]=].
+    1. [=Update all AudioSession states=] of |audioSession|'s [=top-level browsing context=] with |audioSession|.
+    1. For each |element| of |audioSession|.[=AudioSession/[[elements]]=], [=update an element|update=] |element|.
+    1. Let |newType| be the result of [=compute the audio session type|computing the type=] of |audioSession|.
```

Comment:
Could you explain why we need to compute audio session type? The `compute the audio session type` algorithm seems being used to only set the type to a session if a session is active. Any specific reason we want to prohibit setting type on a non-active audio session?

In addition, in `compute the audio session type` algorithm, why it doesn't cover all the types?

Reply:
When type is `auto`, we could just set the audio session type to `auto`.
When activating the audio session as `auto`, it is good to be more precise and state for instance that an `auto` audio session with audible media elements is actually a `playback` audio session.

It is indeed not covering all types, I should add transient solo, and maybe make `ambient` the default type.

Reply:
Sounds good, let's add more types into the section of `compute the audio session type` later. Thanks!